### PR TITLE
made login alert protocol and added segues to loginviewcontroller fro…

### LINKED ISSUE #111: Prevent non-logged in users from accessing the matching chat & profile pages
Instructions:
* Do this work on a new Branch from development
* If users aren't logged in, for now just show a popup on these pages, prompting the users to login and redirecting them to the log in screen
* investigate potential issues with this approach related to navigation and see what you can do about them 
    * e.g. on redirecting to log in screen, what happens when user clicks check menus again? things like that...

### PR DESCRIPTION
**Pull request by sjl1826**
This commit fixes #111

### CODE REVIEW

File: Bruin Bite.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -704,7 +707,6 @@
 				2A03D94820B7AFA7004EFC97 /* HomeProfileStaticTableViewController.swift in Sources */,
 				770CDEB020266B6000658925 /* ActivityLevelBar.swift in Sources */,
 				BD7870A621B6793A00E46A50 /* PresentAlert.swift in Sources */,
-				BF5D97E620B97F490086D451 /* ChatPreview.swift in Sources */,
```
Comment (HirdayGupta) [inline Bruin Bite.xcodeproj/project.pbxproj]: Any idea why this change occurred? Did you delete a file called "ChatPreview.swift" from Xcode?

File: Bruin Bite.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -719,7 +721,6 @@
 				BF0943E420BBFEEF00214200 /* ChatAPI.swift in Sources */,
 				00C2DF0920B678890023CF75 /* Moya.swift in Sources */,
 				2A744895203BBDBC00793AD3 /* MenuCardView.swift in Sources */,
-				00C2DF0D20B678BE0023CF75 /* UserClass.swift in Sources */,
```
Comment (HirdayGupta) [inline Bruin Bite.xcodeproj/project.pbxproj]: Same here?

File: Bruin Bite/Supporting Files/LoginAlertPresentable.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//  Bruin Bite
+//
+//  Created by Samuel J. Lee on 4/22/19.
+//  Copyright © 2019 Dont Eat Alone. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+// To use this protocol, make a segue with identifier "login"
```
Comment (HirdayGupta) [inline Bruin Bite/Supporting Files/LoginAlertPresentable.swift:12]: Fancy! <3 But not entirely sure we want to do it this way.. more on this later but for now this should be fine!

File: Bruin Bite/Supporting Files/LoginAlertPresentable.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+protocol LoginAlertPresentable: class {}
+
+extension LoginAlertPresentable where Self: UIViewController {
+    func presentAlert() {
```
Comment (HirdayGupta) [inline Bruin Bite/Supporting Files/LoginAlertPresentable.swift]: definitely rename this function. Maybe something like `presentNotLoggedInAlert()`?

File: Bruin Bite/View Controllers/ProfileViewController.swift [hunk 5]
```diff
@@ -9,12 +9,17 @@
...
 
-class ProfileViewController: UIViewController, ReadDelegate, AlertPresentable {
+class ProfileViewController: UIViewController, ReadDelegate, AlertPresentable, LoginAlertPresentable {
 
     @IBOutlet weak var profilePic: UIImageView!
     @IBOutlet weak var userName: UILabel!
     @IBOutlet weak var yearMajor: UILabel!
     @IBOutlet var ShortBio: UITextView!
+
+    override func viewDidAppear(_ animated: Bool) {
```
Comment (HirdayGupta) [inline Bruin Bite/View Controllers/ProfileViewController.swift:19]: Looks good, but out of curiosity, why did you choose `viewDidAppear` and not `viewDidLoad`? I would definitely test this out to make sure there aren't infinite alert loops if you keep coming back to the screen.

**Review by HirdayGupta — CHANGES_REQUESTED** [review]
A partial review! Please address these set of comments as I think about whether your proposed approach would be feasible in the long run. Great effort though, very creative - I liked it! 😃

File: Bruin Bite.xcodeproj/project.pbxproj (hunk 1 above)
Comment (sjl1826) [reply Bruin Bite.xcodeproj/project.pbxproj]: Oh no! Let me check that out...

File: Bruin Bite/View Controllers/ProfileViewController.swift (hunk 5 above)
Comment (sjl1826) [reply Bruin Bite/View Controllers/ProfileViewController.swift:19]: I tested it out and there are not infinite alert loops if you keep coming back to the screen. If you think about it, in a `tabBarController`, when you switch to a different view controller it loads and displays and it stays. So, if I were to put it in `viewDidLoad`, my alert would trigger the first time, but it will not trigger all the other times I revisit the view controller. `viewDidAppear` ensures that it'll be triggered any time the view appears.

File: Bruin Bite/Supporting Files/LoginAlertPresentable.swift (hunk 4 above)
Comment (sjl1826) [reply Bruin Bite/Supporting Files/LoginAlertPresentable.swift]: thanks!

**Review by HirdayGupta — APPROVED** [review]
